### PR TITLE
test(crd-generator): ensure type mappings Java to OpenAPIV3Schema

### PR DESCRIPTION
**Pull request by manusa**
## Description

test(crd-generator): ensure type mappings Java to OpenAPIV3Schema

Prerequisite for #5877

/cc @shawkins @andreaTP 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by andreaTP — APPROVED** [review]
Thanks for taking the time to get this together Marc.

In the current state most of the primitive Java types are usable only if annotated with `@PreserveUnknownFields` ...

File: crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Arguments.of("aBooleanObj", "boolean"),
+        // Arguments.of("aChar", "string"), // TODO: Not even present in the CRD
+        Arguments.of("aCharacterObj", "object"), // to review
+        Arguments.of("aCharArray", "array"),
+        Arguments.of("aCharSequence", "object"), // to review
+        Arguments.of("aString", "string"),
+        Arguments.of("aStringArray", "array"),
+        // Arguments.of("aByte", "?"), // TODO: Not even present in the CRD
+        Arguments.of("aByteObj", "object"), // to review
+        Arguments.of("aByteArray", "array"));
```
Comment (shawkins) [inline crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java:87]:
```suggestion
        Arguments.of("aByteArray", "array")); // to review
```

byte[] should default to a base64 encoded string.

**Review by shawkins — COMMENTED** [review]
Looks good, could UUID be included as well?

File: crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java (hunk 1 above)
Comment (shawkins) [reply crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java:87]: @manusa that's why the suggestion is to add the to review comment

Comment (manusa) [reply crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java]: yes, yes, sorry. I realized when adding the uuid field

**Comment by manusa** [comment]
:facepalm: I wasn't accounting for the crd generator to kick in during compilation phase, that's why it fails in 21 regardless of the test annotation :facepalm::facepalm::facepalm:.

I'll try to figure out a simple way to disable this

**Review by shawkins — APPROVED** [review]
LGTM, thanks @manusa

**Pull request merged**